src/lwt_support.ml

```diff
@@ -57,15 +57,26 @@ let with_fd fd ~callback =
     (* The Lwt.catch below prevents errors on double close of the fd. *)
     (fun () -> Lwt.catch (fun () -> Lwt_unix.close fd) (fun _ -> Lwt.return_unit))
 
-let with_open_connection_fd host port ~callback =
-  Lwt_unix.gethostbyname host >>= fun he ->
-  if (Array.length he.Lwt_unix.h_addr_list) = 0
-  then Lwt.fail (Host_not_found host)
-  else
-    let ip = he.Unix.h_addr_list.(0) in
-    let addr = Unix.ADDR_INET(ip, port) in
-    let s = Lwt_unix.socket Lwt_unix.PF_INET Lwt_unix.SOCK_STREAM 0 in
+let with_open_connection_fd addr ~callback =
+  match addr with
+  | Unix.ADDR_INET(inet_addr, port) ->
+    Lwt_unix.gethostbyaddr inet_addr >>= fun he ->
```

Comment:
I think this is redundant now, you can just set `addr = inet_addr`, the hostname lookup and failing with Host_not_found would have to be done by the caller, but AFAICT this was always called with "localhost" so it'd never fail with Host not found.

I think this function can be simplified to:
```ocaml
let with_open_connection_fd addr ~callback =
    let s = Lwt_unix.socket (Unix.domain_of_sockaddr addr) Lwt_unix.SOCK_STREAM 0 in
    with_fd s ~callback:(fun fd -> 
        Lwt_unix.setsockopt fd Lwt_unix.SO_KEEPALIVE true;
        Lwt_unix.connect fd addr >>= fun () ->
        callback fd)
```